release.rb

```diff
@@ -0,0 +1,215 @@
+    github_tags_url = 'https://github.com/jwfriese/Quick/tags'
+    carthage_framework_name = 'Quick'
+    carthage = ENV["CARTHAGE"] || 'carthage'
+    pod = ENV["COCOAPODS"] || 'pod'
+    version_tag = "v#{version}"
+
+    puts 'Verifying Local Directory for Release'
+    carthage_available = verify_availability(carthage) 
+    die("Carthage is required to produce a release. Aborting.") unless carthage_available
+    puts ' > Carthage is installed'
+
+    pod_available = verify_availability(pod)
+    die(" Cocoapods is required to produce a release. Aborting.") unless pod_available
+    puts ' > Cocoapods is installed'
+
+    puts ' > Is this a reasonable tag?'
+
+    verify_version(version_tag, force)
+
+    release_notes = gather_release_notes(release_notes_file_path, version, pod_name)
+    puts " > Release notes: #{release_notes}"
+
+    die("Cannot find podspec: #{pod_spec}. Aborting.") unless File.file?("#{pod_spec}")
+
+    puts ' > Podspec exists'
+    
+    pgp_found = system("git config --get user.signingkey > /dev/null")
+    unless pgp_found
+      puts '[ERROR] No PGP found to sign tag. Aborting.'
+      puts
+      puts '  Creating a release requires signing the tag for security purposes. This allows users to verify the git cloned tree is from a trusted source.'
+      puts '  From a security perspective, it is not considered safe to trust the commits (including Author & Signed-off fields). It is easy for any'
+      puts '  intermediate between you and the end-users to modify the git repository.'
+      puts
+      puts "  While not all users may choose to verify the PGP key for tagged releases. It is a good measure to ensure 'this is an official release'"
+      puts '  from the official maintainers.'
+      puts
+      puts "  If you're creating your PGP key for the first time, use RSA with at least 4096 bits."
+      puts
+      puts 'Related resources:'
+      puts ' - Configuring your system for PGP: https://git-scm.com/book/tr/v2/Git-Tools-Signing-Your-Work'
+      puts ' - Why: http://programmers.stackexchange.com/questions/212192/what-are-the-advantages-and-disadvantages-of-cryptographically-signing-commits-a'
+      puts
+      return
+    end
+
+    puts " > Found PGP key for git"
+
+    #verify_cocoapods_trunk_ownership(pod_name)
+    #puts " > Verified ownership to #{pod_name} pod"
+
+    puts "--- Releasing version #{version} (tag: #{version_tag})..."
+
+    puts "-> Ensuring no differences to origin/#{remote_branch}"
+    fetched_origin = system("git fetch origin")
+    die("Failed to fetch origin. Aborting") unless fetched_origin
+
+    head_aligned_with_origin = system("git diff --quiet HEAD \"origin/#{remote_branch}\"")
+    die("HEAD is not aligned to origin/#{remote_branch}. Cannot update version safely. Aborting.") unless head_aligned_with_origin
+
+    puts "-> Building Carthage release"
+    built_carthage_framework = system("#{carthage} build --no-skip-current --platform iOS,Mac")
```

Comment:
Sounds good.

I can't remember, though: didn't we at one point attach Carthage binaries to all of our releases? Looking at past Quick and Nimble releases now, though, I can't find any. GitHub's audit logs don't go back far enough for me to see all the activity that's ever been done around our releases, but I assume that at one point we deleted the binaries. Perhaps because Swift is not ABI stable? If anyone in @Quick/core knows for sure, feel free to chime in.

Anyway, since it appears we don't attach Carthage binaries, maybe we can delete this code for now?